### PR TITLE
[REEF-366]: Initial project setup for Vortex

### PR DESCRIPTION
**Pull request by johnyangk**
This addressed the issue by:
  * Creating a directory 'lang/java/reef-applications/vortex'
  * Creating necessary pom files
  * Adding 'reef-applications' submodule to the pom file at the root directory

JIRA: [REEF-366](https://issues.apache.org/jira/browse/REEF-366)

### CODE REVIEW

**Comment by bgchun** [comment]
Looks good to me. :+1: 
I will wait for others to take a look.

**Comment by markusweimer** [comment]
This adds one new project, `reef-applications`. If we expect additional ones in the future, shouldn't we make `reef-applications` a parent project with `reef-vortex` as a submodule?

**Comment by johnyangk** [comment]
@markusweimer I see, so are you suggesting we have a parent `pom.xml`, which can contain common dependencies such as `reef-common`, in `reef-applications` and then each application(e.g. `reef-vortex`) have its own `pom.xml`?

**Comment by johnyangk** [comment]
Made changes to address @markusweimer's comments.
@markusweimer, could you please review it again?

File: lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/VortexDriver.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          .setMemory(64)
+          .setNumberOfCores(1)
+          .build());
+    }
+  }
+
+  public final class EvaluatorAllocatedHandler implements EventHandler<AllocatedEvaluator> {
+    @Override
+    public void onNext(final AllocatedEvaluator allocatedEvaluator) {
+      LOG.log(Level.INFO, "Evaluator Allocated: {0}", allocatedEvaluator);
```
Comment (markusweimer) [inline lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/VortexDriver.java]: This is a resource leak. I know this is a placeholder, but please call `allocatedEvaluator.close()` here :-)

**Comment by markusweimer** [comment]
This looks good to me, but for the small issue I mentioned above. Let's see what people in PST say.

File: lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/VortexDriver.java (hunk 1 above)
Comment (johnyangk) [inline lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/VortexDriver.java]: Fixed. Thanks for the catch!

**Comment by markusweimer** [comment]
+1 from me

**Comment by bgchun** [comment]
If I don't hear any concern by tomorrow, I will merge it.

**Pull request closed without merging**